torch/csrc/jit/runtime/argument_spec.h

```diff
@@ -223,8 +224,8 @@ struct CompleteArgumentInfoPOD {
   unsigned type : 8; // scalar type
   unsigned defined : 1;
   unsigned requires_grad : 1;
-  signed device : 14;
-  unsigned dev_type : 16;
+  unsigned dev_type : 14;
```

Comment:
You're reducing dev_type from 16 to 14 here!
Also we can keep the current order, since bitfield gets nicely packed anyways

Reply:
I had to steal 2 bits from somewhere as the whole struct needs to fit into 64 bit. `DeviceType` is an `int8_t` anyway (with only 21 types being in use), so I don't think it matters.

Reply:
In my local environment, the static assert is triggered when I swap `device` and `dev_type` back.

Reply:
In that case, you you make it `signed dev_type: 8;` to match the reality of things?